hashicorp/dev-portal
# feat: add bookmark auth cta

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-ksadd-bookmark-auth-cta-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1202520168081556/1202791375540708) 🎟️

## 🗒️ What

This PR adds the pop up dialog for unauthenticated users, prompting them to sign-in in order to save the bookmark. It also refactors things so that both the tutorial meta and bookmark card icon can use this dialog via an HOC called `withDialog`. We can decide later whether this HOC will also handle the data handling based on user interaction. 

## 📸 Design Screenshots

<img width="717" alt="Screen Shot 2022-08-17 at 3 19 53 PM" src="https://user-images.githubusercontent.com/36613477/185253678-4ec80ec9-664e-44c3-b3c9-fa54efa442a5.png">


## 🧪 Testing
- Visit a page with a [bookmark button](https://dev-portal-git-ksadd-bookmark-auth-cta-hashicorp.vercel.app/terraform/tutorials)
- Hit the button, you should see the sign in prompt dialog. 

To test the full flow, run this locally
- Do the same steps and you should be able to sign in. If you click the bookmark icon when signed in, nothing should happen. 

## 💭 Anything else?

This is just a stub and still needs to be integrated with the work in [#838](https://github.com/hashicorp/dev-portal/pull/838)
